cranelift-codegen/src/regalloc/diversion.rs

```diff
@@ -163,9 +177,94 @@ impl RegDiversions {
         self.current.remove(&value).map(|d| d.to)
     }
 
+    /// Reset the current diversion for the entry of an `Ebb`, based on the recorded content
+    /// available on `func.entry_diversions`.
```

Comment:
This comment is kind of literal. Maybe better: "Resets the state of the current diversion to the recorded EntryDiversions on the given ebb`.